addons/account/models/account_invoice.py

```diff
@@ -560,8 +562,21 @@ def _prepare_tax_line_vals(self, line, tax):
             'account_analytic_id': tax['analytic'] and line.account_analytic_id.id or False,
             'account_id': self.type in ('out_invoice', 'in_invoice') and (tax['account_id'] or line.account_id.id) or (tax['refund_account_id'] or line.account_id.id),
         }
+        # If the taxes generate moves on the same financial account as the invoice line,
+        # propagate the analytic account from the invoice line to the tax line.
+        # This is necessary in situations were (part of) the taxes cannot be reclaimed,
+        # to ensure the tax move is allocated to the proper analytic account.
+        if not val.['account_analytic_id'] and line.account_analytic_id and val['account_id'] == line.account_id.id:
+            val['account_analytic_id'] = line.account_analytic_id.id
         return vals
 
+    def _cleaned_tax_grouped(self, tax_grouped):
```

Comment:
From my standpoint, I have an interest, that many localisators in SA take this approach because this makes their solution more compatible with ours and therefore, executing a migration or serving a support case is more easy... It's half code, half politics 😸 I understand if Odoo is not willing to prescribe this, but there is an ecosystem benefit of such explicit hints... If an empty hook canot be accepted, then second best option would be an explicit comment, what do you think?